src/ctx.ts

```diff
@@ -62,6 +62,11 @@ export class Ctx {
       initializationOptions: { clangdFileStatus: true },
       disableDiagnostics: this.config.disableDiagnostics,
       outputChannel,
+      middleware: {
+        provideOnTypeFormattingEdits: (document, position, _ch, options, token, next) => {
```

Comment:
Can we add a comment like:
coc sends "\n" when exiting insert mode, when there is no newline added to the doc.